assets/src/edit-story/app/story/useStoryReducer/reducers/deleteElements.js

```diff
@@ -81,9 +81,16 @@ function deleteElements(state, { elementIds }) {
     newElements = [oldPage.defaultBackgroundElement, ...newElements];
   }
 
+  // Remove animations associated with elements
+  const oldAnimations = oldPage.animations || [];
+  const newAnimations = oldAnimations.filter((anim) =>
+    anim.targets.some((elementId) => !validDeletionIds.includes(elementId))
+  );
```

Comment:
Or is this written under the assumption that an animation will only have 1 target in its target array?

Reply:
Yeah, animations will only have 1 target (so we'll have to update our templates to split out animations with multiple targets).  This is because when we do group animations, we still want the ability to apply different duration/delays to each one sooo... the easiest way is to just keep everything at 1-to-1.

Reply:
yea that makes sense. Honestly might be easiest to just write a quick node script to go through and flatten all the animations in the templates out

Reply:
Yeah, that and we'll have to update our story-anim-tool so when we apply animations to groups of elements, it creates a separate animation record for each one.